core/sail/fts/lucene-spin/src/test/java/org/eclipse/rdf4j/lucene/spin/LuceneSailSpinTest.java

```diff
@@ -37,100 +40,114 @@
+import org.junit.rules.TemporaryFolder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * This test class reproduces errors described in issues #220 and #235.
+ * This test class reproduces errors described in issues #220 and #235 with simplified IDE (#739).
  * 
  * @see <a href="https://github.com/eclipse/rdf4j/issues/220">issue #220</a>
  * @see <a href="https://github.com/eclipse/rdf4j/issues/235">issue #235</a>
+ * @see <a href="https://github.com/eclipse/rdf4j/issues/739">issue #739</a>
  */
-public abstract class AbstractLuceneSailSpinTest {
+public class LuceneSailSpinTest {
 
 	private static final String DATA = "org/eclipse/rdf4j/sail/220-example.ttl";
 
-	private static Logger log = LoggerFactory.getLogger(AbstractLuceneSailSpinTest.class);
+	private static Logger log = LoggerFactory.getLogger(LuceneSailSpinTest.class);
+
+	@ClassRule
+	public static TemporaryFolder tempDir = new TemporaryFolder();
 
 	private Repository repository;
 
 	private RepositoryConnection connection;
 
-	private SearchIndex searchIndex;
-
 	@Before
 	public void setUp()
 		throws Exception
 	{
+
+		//activate sail debug mode
+		// System.setProperty("org.eclipse.rdf4j.repository.debug", "true");
 		// load data into memory store
 		MemoryStore store = new MemoryStore();
 
 		// add Support for SPIN function
 		SpinSail spin = new SpinSail(store);
-		spin.setEvaluationMode(TupleFunctionEvaluationMode.TRIPLE_SOURCE);
-
-		// add Lucene support
-		Properties parameters = new Properties();
-		configure(parameters);
-		searchIndex = LuceneSail.createSearchIndex(parameters);
-		spin.addQueryContextInitializer(new SearchIndexQueryContextInitializer(searchIndex));
-		repository = new SailRepository(spin);
+
+		// add Lucene Spin Sail support
+		LuceneSpinSail luc = new LuceneSpinSail(spin);
+		File tmpDirFolder = tempDir.newFolder();
+		log.debug("data file: {}", tmpDirFolder);
+		luc.setDataDir(tmpDirFolder);
+		repository = new SailRepository(luc);
+
+		// set up parameters
+		configure(luc.getParameters());
+
 		repository.initialize();
+		// local connection used only for population
+		try (RepositoryConnection localConn = repository.getConnection()) {
+			localConn.begin();
+			populate(localConn);
+			localConn.commit();
+		}
 
-		connection = repository.getConnection();
-		populate(connection, searchIndex);
+		// local connection for verification only
+		try (RepositoryConnection localConn = repository.getConnection()) {
+			// validate population
+			//localConn.begin();
+			int count = countStatements(localConn);
+			log.trace("storage contains {} triples", count);
+			assert count > 0;
+			//localConn.commit();
+			localConn.close();
+		}
 
-		// validate population
-		int count = countStatements(connection);
-		log.info("storage contains {} triples", count);
-		assert count > 0;
+		// testing connection
+		connection = repository.getConnection();
+		connection.begin();
+		assert connection.isActive() : "connection is not active";
```

Comment:
The issue with using assert rather than the JUnit assertTrue, is that assert is only tested if the JVM is running in server mode. This potentially creates the possibility of this test passing on one JVM but failing in another. Best just to use the JUnit assertTrue/etc. to avoid those issues. Also need to fix other uses of assert in this test.